lib/yauzl-ts/RandomAccessReader.ts

```diff
@@ -1,96 +1,112 @@
-import EventEmitter from "node:events";
-import {PassThrough, Readable, Writable} from "stream";
-import {RefUnrefFilter} from "./internal/RefUnrefFilter";
-import {AssertByteCountStream} from "./internal/AssertByteCountStream";
+import EventEmitter from 'node:events'
+import { PassThrough, Writable } from 'stream'
+import type { Readable, Transform } from 'stream'
+
+import { AssertByteCountStream } from './internal/AssertByteCountStream'
+import { RefUnrefFilter } from './internal/RefUnrefFilter'
+
+export interface RandomAccessReaderCreateReadStream {
+  start: number
+  end: number
+}
 
 export class RandomAccessReader extends EventEmitter {
-    private refCount: number;
+  private refCount: number
+
+  constructor() {
+    super()
+    this.refCount = 0
+  }
+
+  ref() {
+    this.refCount += 1
+  }
+
+  unref() {
+    this.refCount -= 1
+
+    if (this.refCount > 0) return
+    if (this.refCount < 0) throw new Error('invalid unref')
+
+    this.close((err?: Error) => {
+      if (err) return this.emit('error', err)
+
+      this.emit('close')
+    })
+  }
+
+  createReadStream(options: RandomAccessReaderCreateReadStream): Transform {
+    const start = options.start
+    const end = options.end
+
+    if (start === end) {
+      const emptyStream = new PassThrough()
+      setImmediate(function () {
+        emptyStream.end()
+      })
+      return emptyStream
+    }
+    const stream = this._readStreamForRange(start, end)
+
+    let destroyed = false
+    const refUnrefFilter = new RefUnrefFilter(this)
+    stream.on('error', function (err) {
+      setImmediate(function () {
+        if (!destroyed) refUnrefFilter.emit('error', err)
+      })
+    })
+    refUnrefFilter.destroy = function (_?: Error) {
+      stream.unpipe(refUnrefFilter)
+      refUnrefFilter.unref()
+      stream.destroy()
+
+      return this
+    }
+
+    const byteCounter = new AssertByteCountStream(end - start)
+    refUnrefFilter.on('error', function (err) {
+      setImmediate(function () {
+        if (!destroyed) byteCounter.emit('error', err)
+      })
+    })
+    byteCounter.destroy = function () {
```

Comment:
it's a bit of a code smell to inject new methods into a class. Can we make `destroy` a permanent method and pass necessary parameters via fields on an object instead?

doesn't have to be done in this PR, just a ToDo would suffice